tests/unit/amo/components/TestInstallWarning.js

```diff
@@ -206,74 +248,104 @@ describe(__filename, () => {
+
+      render({
+        _couldShowWarning,
+        _tracking,
+        addon,
+        variant,
       });
 
-      it('does not send the event on mount if couldShowWarning is false', () => {
-        const _couldShowWarning = sinon.stub().returns(false);
+      sinon.assert.calledWith(_tracking.sendEvent, {
+        action: variant,
+        category: EXPERIMENT_CATEGORY_DISPLAY,
+        label: addon.name,
+      });
+    });
 
-        render({ _couldShowWarning, _tracking, variant });
+    it('does not send the event on mount if couldShowWarning is false', () => {
+      const _couldShowWarning = sinon.stub().returns(false);
 
-        sinon.assert.notCalled(_tracking.sendEvent);
+      render({
+        _couldShowWarning,
+        _tracking,
+        variant: VARIANT_INCLUDE_WARNING_CURRENT,
       });
 
-      it('does not send the event on mount if a variant is not set', () => {
-        const _couldShowWarning = sinon.stub().returns(true);
+      sinon.assert.notCalled(_tracking.sendEvent);
+    });
+
+    it('does not send the event on mount if a variant is not set', () => {
+      const _couldShowWarning = sinon.stub().returns(true);
+
+      render({ _couldShowWarning, _tracking, variant: undefined });
 
-        render({ _couldShowWarning, _tracking, variant: undefined });
+      sinon.assert.notCalled(_tracking.sendEvent);
+    });
+  });
+
+  it.each([VARIANT_INCLUDE_WARNING_CURRENT, VARIANT_INCLUDE_WARNING_PROPOSED])(
+    'displays a warning if couldShowWarning is true and the user is in the %s branch',
+    (variant) => {
+      const _couldShowWarning = sinon.stub().returns(true);
 
-        sinon.assert.notCalled(_tracking.sendEvent);
+      const root = render({
+        _couldShowWarning,
+        variant,
       });
+      expect(root.find(Notice)).toHaveLength(1);
+    },
+  );
+
+  it('does not display a warning if couldShowWarning is false', () => {
+    const _couldShowWarning = sinon.stub().returns(false);
+
+    const root = render({
+      _couldShowWarning,
+      variant: VARIANT_INCLUDE_WARNING_CURRENT,
     });
+    expect(root.find(Notice)).toHaveLength(0);
   });
 
-  it('displays a warning if couldShowWarning is true and the user is in the included cohort', () => {
+  it('does not display a warning if the user is in the excluded cohort', () => {
     const _couldShowWarning = sinon.stub().returns(true);
 
     const root = render({
       _couldShowWarning,
-      variant: VARIANT_INCLUDE_WARNING,
+      variant: VARIANT_EXCLUDE_WARNING,
     });
-    expect(root.find(Notice)).toHaveLength(1);
+    expect(root.find(Notice)).toHaveLength(0);
   });
 
-  it('does not display a warning if couldShowWarning is false', () => {
-    const _couldShowWarning = sinon.stub().returns(false);
+  it('sets the expected notice type and message for VARIANT_INCLUDE_WARNING_CURRENT', () => {
+    const _couldShowWarning = sinon.stub().returns(true);
 
     const root = render({
```

Comment:
There are a couple tests like this that are testing functionality with a warning shown but they aren't concerned with *how* the component decided to show the warning. It might be a bit easier to wrap those up in a render helper rather than definining `_couldShowWarning = () => true` every time. I see that you already have `renderWithWarning()` but that's a little different so maybe you want `renderWithWarningEnabled()`, etc. Just something to consider.

Reply:
Good point. I refactored it a bit so I could use `renderWithWarning` for both cases, rather than creating an extra `renderWithWarningEnabled`.